### ISSUE TITLE
Overflow trait and job preferences issues.

### ISSUE THREAD

**Issue by Pinkufeck** [tgstation/tgstation#72719]
>Want to play a round
>Don't check preferences, already set, with assistant off
>You spawn, and are forced into whatever is the overflow
>You are forced into it if your preferences for that job was anything above "none"
>Find yourself in a job your didn't necessarly want (on low) or that can't be done with so many people. 

Why does overflow jobs not inherit their preferences from assistant, to which it's supposed to be a replacement ?

**Comment by Serxule** [comment]
> > Want to play a round
> > Don't check preferences, already set, with assistant off
> > You spawn, and are forced into whatever is the overflow
> > You are forced into it if your preferences for that job was anything above "none"
> > Find yourself in a job your didn't necessarly want (on low) or that can't be done with so many people.
> 
> Why does overflow jobs not inherit their preferences from assistant, to which it's supposed to be a replacement ?

overflow takes the slot of "join as assistant if preferences are unavailable" which means you get forced into whatever said overflow job is, preferences dont matter if you have said option selected and no jobs you have selected are open

**Comment by Mothblocks** [comment]
I've read this post like 6 times and still don't understand what you're saying, can you rephrase it?

**Comment by Pinkufeck** [comment]
Assistant : off
Overflow for let's say, geneticist
You have something on high
You won't get what's on high, because pref for geneticist changed from low to yes (because it's the overflow)
You get forced into a job you didn't want because it doesn't inherent preference from assistant (here; no)

**Comment by Mothblocks** [comment]
Oh I see

**Comment by Mothblocks** [comment]
I think that's the point to be fair

**Comment by Serxule** [comment]
> Assistant : off Overflow for let's say, geneticist You have something on high You won't get what's on high, because pref for geneticist changed from low to yes (because it's the overflow) You get forced into a job you didn't want because it doesn't inherent preference from assistant (here; no)

oh, 99% sure thats intentional since having a job selected means you want to play that job (to some degree), and since overflow becomes a yes/no its automatically set to yes

**Comment by Pinkufeck** [comment]
Guh.
